### DISCUSSION TITLE
rounded button still detects mouse outside?

### DISCUSSION THREAD

**Discussion by Neo-Mind** [mikke89/RmlUi#534] [Q&A] [answered]
Is the client area always rectangular in RML? 
I tried creating a round button (with border-radius set to approximately half the width), but the element is still detecting hover & clicks just outside it's borders. In HTML, this is not the case. 

Anyway to fix this in RML/RCSS itself or will I need a custom element for it?

**Comment by mikke89** [comment, marked as answer]
Yeah, we currently only do a hit test against the rectangular area of the element. Certainly something that could be improved if someone is up to the task.

**Comment by Neo-Mind** [comment]
Ok thanks for confirming.  For now, I modified the ``IsPointWithinElement`` function to make it work. 
I am picking up the radii from computed values and using them in the circle equations at the respective corners to cross check against the supplied point.

**Reply by mikke89** [reply]
Nice, pull request? :)

Be aware that a completely general solution need to handle situations where the border corners become elliptic. You can see some examples of this in the visual tests, and also [in the documentation](https://mikke89.github.io/RmlUiDoc/pages/rcss/colours_backgrounds.html#border-radius).
